agent/version/version_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package version_test
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+
+	"github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/agent/version"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Agent version", func() {
+
+	Context("When the version number and date are not set", func() {
+
+		It("Leaves the version and date fields empty in response", func() {
+			expected := version.Info{
+				Major:     "",
+				Minor:     "",
+				Patch:     "",
+				BuildDate: "",
+				GoVersion: runtime.Version(),
+				Compiler:  runtime.Compiler,
+				Platform:  fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+			}
+			Expect(version.Get()).Should(Equal(expected))
+		})
+	})
+
+	Context("When only date is set", func() {
+		BeforeEach(func() {
+			date, err := exec.Command("date").Output()
+			Expect(err).NotTo(HaveOccurred())
+			version.BuildDate = string(date)
+		})
+
+		AfterEach(func() {
+			version.BuildDate = ""
+		})
+
+		It("Leaves version field empty in response", func() {
+			expected := version.Info{
+				Major:     "",
+				Minor:     "",
+				Patch:     "",
+				BuildDate: version.BuildDate,
+				GoVersion: runtime.Version(),
+				Compiler:  runtime.Compiler,
+				Platform:  fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+			}
+			Expect(version.Get()).Should(Equal(expected))
+		})
+	})
+
+	Context("When version is set", func() {
```

Comment:
Suggestion: Should we have one more context under this for `when the git tag is not proper` for e.g. 1.0, etc.